server/player/player.go

```diff
@@ -90,11 +90,13 @@ type Player struct {
 	breakParticleCounter atomic.Uint32
 
 	hunger *hungerManager
+
+	provider Provider
 }
 
 // New returns a new initialised player. A random UUID is generated for the player, so that it may be
 // identified over network.
-func New(name string, skin skin.Skin, pos mgl64.Vec3) *Player {
+func New(name string, skin skin.Skin, pos mgl64.Vec3, data *Data) *Player {
```

Comment:
Does this data need to be a pointer?

Reply:
My reasoning to make the data a pointer is that players could pass nil if they want to use default values, for example since it seems you can make a player without a session I can imagine people might want to use the default values and when a new player joins they dont have any data.

An alternative to this is I could have an exported function like `player.DefaultData()` or something like that and instead pass that on if no data was found.

Reply:
I asked around and this should be fine according to Sander so ill leave this one as resolved 😄